app/routes.js

```diff
@@ -1,8 +1,13 @@
 // const controller = require('./controllers/controller');
 const { healthCheck } = require('./controllers/healthCheck');
+const { singUp } = require('./controllers/users.controller');
+const { validateUserSchema } = require('./middlewares/user_schema');
+const { validate } = require('./middlewares/validate_request');
+const { emailExists } = require('./middlewares/user');
 
 exports.init = app => {
   app.get('/health', healthCheck);
+  app.post('/users', [validateUserSchema, validate, emailExists], singUp);
```

Comment:
Here I think there are too many middlewares. For schema validation 1 should be enough, and for data validation it would be better to have the code in the controller/interactor and not as a middleware

Reply:
I have removed the middleware to validate the email and use controller for this.